kedro/io/partitioned_data_set.py

```diff
@@ -220,6 +226,7 @@ def _list_partitions(self) -> List[str]:
             path
             for path in self._filesystem.find(self._normalized_path, **self._load_args)
             if path.endswith(self._filename_suffix)
+            and bool(re.search(self._regex_filter, Path(path).name))
```

Comment:
1. For a small performance improvement we should probably [compile](https://docs.python.org/3/library/re.html#re.compile) the expression before using it for filtration
2. Shall we maybe apply regex filter to the whole `path` rather than just its filename? I think it will unlock more use cases for the users

Reply:
Thank you @DmitriiDeriabinQB , about the second comment. I am only applying the filter to the files in the path because `path` the same for al files, am I right?